pkg/controller/certificates/controller.go

```diff
@@ -262,3 +271,24 @@ func (cc *CertificateController) maybeAutoApproveCSR(csr *certificates.Certifica
 	})
 	return cc.kubeClient.Certificates().CertificateSigningRequests().UpdateApproval(csr)
 }
+
+func (cc *CertificateController) sign(csr *certificates.CertificateSigningRequest) ([]byte, error) {
+	cc.signerLock.Lock()
+	defer func() {
+		cc.signer.SetPolicy(onlySigningPolicy)
+		cc.signerLock.Unlock()
+	}()
+	var usages []string
+	for _, usage := range csr.Spec.Usages {
+		usages = append(usages, string(usage))
+	}
+	cc.signer.SetPolicy(&config.Signing{
+		Default: &config.SigningProfile{
+			Usage:        usages,
+			Expiry:       helpers.OneYear,
+			ExpiryString: "8760h",
```

Comment:
All of this should be inputs to the certificate controller.  I think the sign method should actually be a separate interface passed into this controller that hides the implementation details from the controller.  Same for maybeApprove (I'm opening a PR to do that right now).